api/src/main/java/org/apache/iceberg/view/UpdateViewRepresentations.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.util.List;
+import java.util.function.Predicate;
+import org.apache.iceberg.PendingUpdate;
+
+/**
+ * API for updating view representations.
+ *
+ * <p>Apply returns the updated view representations as a list for validation.
+ *
+ * <p>When committing, these changes will be applied to the current view metadata. Commit conflicts
+ * will be resolved by applying the pending changes to the new view metadata.
+ */
+public interface UpdateViewRepresentations extends PendingUpdate<List<ViewRepresentation>> {
```

Comment:
I'm not confident that this is how we want to evolve a view. Like I said above, I don't like exposing a big complex object (`ViewRepresentation`) to users.

I'd prefer to start off with just `updateDefinition` or similar that allows replacing the current view version. We can adapt that to add representations implicitly. That update can contain a call to both `withQuery` or a future `withSubstrait` or something that creates parallel representations.

Reply:
Sounds good. We can add this or something similar when the use cases come up.